src/Scheduler.php

```diff
@@ -0,0 +1,155 @@
+<?php
+namespace Riskio\EventScheduler;
+
+use DateInterval;
+use DateTime;
+use Riskio\EventScheduler\TemporalExpression\TemporalExpressionInterface;
+use Traversable;
+
+class Scheduler implements SchedulerInterface
+{
+    /**
+     * @var array
+     */
+    protected $scheduledEvents = [];
+
+    /**
+     * @var DateRange
+     */
+    protected $dateRange;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function schedule(
+        SchedulableEvent $event,
+        TemporalExpressionInterface $temporalExpression
+    ) {
+        if ($this->isScheduled($event)) {
```

Comment:
one may want to schedule an event more than once using different temporal expression. So either you don't add this check or you add the temporal expression as the second parameter